### ISSUE TITLE
Information: Evo Documentation

### ISSUE THREAD

**Issue by modxuser** [modxcms/evolution#1106]
It has been brought to attention that the Evo documentation has lots of missing and outdated content as well as some of it being misleading / confusing.

I have just requested access to the docs to be able to update where possible - **BUT**, without community feedback it won't happen :)

I have a copy of the original documentation from 0.9.1 and I will cross reference and update where possible

Note that the Evo documentaion has pretty much all of the images missing, so they will need reproducing

I would personally prefer to do this on an Evo community driven site, but I don't think that is currently available, or am I wrong here @dmi3yy 

If anyone has suggestions for updating the docs please let us know. Help will be needed.

**Comment by modxuser** [comment]
Update:

5 days later and I am still waiting on the LLC

**Comment by modxuser** [comment]
This obviously seem to be a waste of time - it's now almost a month since submitting the form and still no answer.

I will try and contact someone from the LLC directly, but I don't give it much chance of success

**Comment by Deesen** [comment]
I think there is no need. It seems the final decision has been made. At release of MODX3 the Evo-support from MODX (and MODX.com) will be dropped officially. So we would need to create our own fork and omit the name MODX.

**Comment by bossloper** [comment]
A new name will be needed then. Thinks hard .... .... how about ... er ClipperCMS ... ouch! ;-)

**Comment by modxuser** [comment]
They officially dropped all support Evo a long time ago Dennis and whether MODX3 is ever going to be a reality is up in the sky, if you ask me. I twas supposed to have been released in 2014 if I remember rightly.

@bossloper couldn't stop laughing after reading that :)

**Comment by Cipa** [comment]
How about xmod? :)

**Comment by modxuser** [comment]
My opinion:

I see no reason to change the name, just drop MODX and leave it as Evo or Evolution

And with the drop of the word "MODX" the code should be changed as such, no-longer $modx etc. but something else, e.g. $evo

Also goes for the text in many files saying "please use the modx manager", the folder manager should also be changed - more seperation from Revo maybe ?

**Comment by bossloper** [comment]
Whatever the name, could the name be stored in the system_settings table, lang file or config.inc.php and then all occurrences changed to reference that single setting... then easy to change if/as and when?

**Comment by modxuser** [comment]
Update:

I have been working on a copy of the current documentation since Friday and I am still banging my head on the desk asking myself why I started it.

There are tons of errors in it and loads of images and or files are missing.

Once finished, it will be a "standalone version", but finished is a big word.

I have interlinked all of the HTML files (106), cleaned up all the code by removing various scripts, HTML and general rubbish that's not needed.

Current state of play - going through all pages and finding images, correcting paths and listing all images that are missing.

Once all files are processed I will upload the docs as a zip file - but there will still be images missing and some dead links - I don't want to remove the dead links as of yet, as they are a reference to what is missing - remove them and you don't know they are missing - simple logic.

Note: i have replaced "MODx" with "((CMSNAME))" for faster "find and replace" as I am going through page for page

If anyone has any ideas on how to take this further - feel free :)

**Comment by Deesen** [comment]
Thanks for your efforts! I have no clue how which system to use to make an official online documentation like http://docs.evolution-cms.com but we need such a platform ASAP so every effort can be put directly there.

@dmi3yy What do you think? Can we create everything necessary for a new version of documentation like above link?

**Comment by yama** [comment]
I love the design of Nicola. I hope for him if we will make a site someday!
I would like to continue a good relationship with LLC. I respect Revo, an excellent sibling.

**Comment by modxuser** [comment]
Just an update:

I have successfuly been in contact with Jay Gilmore from the LLC and he has now added 1.2.1 to the blog and updated the download on the MODX site.

He also found that my initial request for the docs was somehow not directed to hime and as such, has asked I f I still want access.

I am in two minds - I have spent a total of about 3 working days on the copy I have and to update the live version with all of the changes would not be a solution that I can provide - I simply no-longer have the time. Another problem is that I have made major changes by renaming etc. which would probably not be accepted or possibly simply not be possible - I am not sure how the live system works

I will continue as is - will get what I have finished to a state that I can zip up and then we will have to see what the future brings - if the LLC do drop all relations to Evo, then I see no point in updating the Docs on the MODX site, as all the work will have to be re-created anyway on a new site

If anyone has any ideas, please let me know :)

**Comment by zabudkin** [comment]
May be you make simple site with your new documentation?

**Comment by modxuser** [comment]
@yama 

Where would you expect to find "[Cross references](https://github.com/modxcms/evolution/wiki/Cross-references)" in the documentation, under "[Content Editing](https://docs.modx.com/evolution/1.0/content-editing)" or "[Designing](https://docs.modx.com/evolution/1.0/designing)"

The same goes for the modifiers found [here](https://github.com/modxcms/evolution/wiki/@IF-@ELSEIF-@ELSE-@ENDIF) and [here](https://github.com/modxcms/evolution/wiki/Global-var)

Cheers

**Comment by modxuser** [comment]
For the record - the DOCs are ready for zipping

From the original 73 missing images, there are now only 2 missing

The majority of the images need updating to replicate the new and improved back-end

Credit where credit is due - I found some of the images on archive.org after chatting with Ryan & Jay from the LLC, the rest I found whilst cross referencing all of the old docs I have.

Some pages have been updated, some pages that had no information at all now have some. I did the best I could from my working experience, at least they are no-longer empty.

Once @yama gets back to me ref. the above post, I will then add that content and zip it all.

I have also added various other files that I have collected over the years, in a separate folder called "Read Me", they contain tutorials and other documentation for Evo e.g "The almost Complete Guide to Constructing Menus in MODx Using Wayfinder", "The Developer's Guide", "How to duplicate a document" etc.

**Comment by yama** [comment]
@modxuser 
Sorry and thanks, I'm pretty busy this month. Please use whatever I wrote contents are freely, I will make various things again when I have the time.

**Comment by modxuser** [comment]
Hi @yama 

You have misunderstood :)

Where would you like the information to be added - in which section of the documentation ?

**Comment by modxuser** [comment]
@Deesen @pmfx @fourroses666 @bossloper @Nicola1971 

Anyone want to give a crack at creating the RTFM side navigation with WayFinder ?

I've given up - 4 hours and couldn't get it to work

If no-one is interested I will simply re-build it using a different method - something easier and better, less code, less classes - basically, less everything LOL

To note is - the present menu is also controlled by JavaScript, If I change the build method, the one I will build wont have that function

Currently all pages are listed in every single page - a lot to load every page

More infos - simply ask here :) All and any help appreciated

p.s The Docs are to 85% in Evo 1.2.1, I am simply cleaning up at the moment and hope to have it finished by next week and the get the menu sorted.

**Comment by Nicola1971** [comment]
@modxuser sorry missed this discussion.
First, Thanks a lot for your hard work.

> Anyone want to give a crack at creating the RTFM side navigation with WayFinder ?

Not understand, Need a template with a sidenav for Evo? or just the sidenav in wayfinder?

**Comment by modxuser** [comment]
@Nicola1971 
just the sidebar Nav for wayfinder - "wayfinder call + custom chunks"

**Comment by fourroses666** [comment]
Is this something? Don't know if you have a BS template.
http://jsfiddle.net/ivangrs/v3ajg1wx/

**Comment by modxuser** [comment]
@fourroses666 

Hi Marc, I have replicated the docs site "1 for 1"

The local version uses 100% the same design / code + files. This was for speed and usability - but I learnt the hard way - it would have been quickers to start from scratch

The only thing I personally can't replicate to 100%, is how the sidebar menu works - the functionality

I asked Jay Gilmore for the Wayfinder call, but they haven't used Wayfinder - they used a combination of multiple snippets / code and javascript for secondary functionality

At the moment, the code "as is" shows all menu items in all pages. That is over 100 sidebar links in every page - which in my opinion is OTT

I may just leave it as is and if at a later stage someone wants to update the layout, it can be done then.

If someone can come up with an idea, maybe using modifiers in conjunction with Wayfinder, then great, but that is out of my technical reach and to be honest, I simply dont have the time to keep testing various solutions.

Evo is a community project and as such I have called on the communtiy to help - if no-one can, no stress whatsoever - I will re-build the nav a different way :)

**Comment by fourroses666** [comment]
Hi , correct me if I'm wrong:

So you have a copy of; https://docs.modx.com/evolution/1.0 ?
And imported the pages into Evo 1.2.1?
And need the correct Wayfinder call for it?

**Comment by dmi3yy** [comment]
We are have goof start for all docs

http://docs.evolution-cms.com
https://github.com/evolution-cms/docs/

but need do translate and update


От 20 января 2017 г. в 14:58:55, Marc (notifications@github.com) написал:

Hi , correct me if I'm wrong:

So you have a copy of; https://docs.modx.com/evolution/1.0 ?
And imported the pages into Evo 1.2.1?
And need the correct Wayfinder call for it?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/modxcms/evolution/issues/1106#issuecomment-274066443>,
or mute the thread

**Comment by modxuser** [comment]
Yes to all :)

![docs-1 21](https://cloud.githubusercontent.com/assets/1859868/22150414/35c9c460-df19-11e6-96e1-031ee8950e58.jpg)

**Comment by fourroses666** [comment]
Ok, I can have a look if I can nail it.

**Comment by modxuser** [comment]
@dmi3yy 
That's great, but I don't know who could translate and update all that - wow, its a lot :)

**Comment by modxuser** [comment]
@fourroses666 I am not sure if @Nicola1971 is giving it a go - maybe if you have a quick chat with him ?

**Comment by modxuser** [comment]
If anyone wants the current system - I can zip it and post here, you can then upload to your local server and test - just a n idea

**Comment by modxuser** [comment]
I am working on getting the docs finished ASAP, so the call is all I need, and if any changes are made to CSS etc. those files will also be needed

**Comment by fourroses666** [comment]
Does this work?

```
[!Wayfinder? 
	&startId=`0` 
	&level=`5` 
	// &removeNewLines=`1`
	&outerTpl=`@CODE:<ol class="ug-toc">[+wf.wrapper+]</ol>`
	&rowTpl=`@CODE:<li[+wf.classes+]><a href="[+wf.link+]" [+wf.attributes+]>[+wf.linktext+]</a>[+wf.wrapper+]</li>`
	&innerTpl=`@CODE:<ol class="ug-toc">[+wf.wrapper+]</ol>`
	&innerRowTpl=`@CODE:<li[+wf.classes+]><a href="[+wf.link+]">[+wf.linktext+]</a></li>`
	&parentRowTpl=`@CODE:<li><i class="icon-caret-right"></i><a href="[+wf.link+]">[+wf.linktext+]</a>[+wf.wrapper+]</li>`
	&activeParentRowTpl=`@CODE:<li class="current"><i class="icon-caret-right"></i><a href="[+wf.link+]">[+wf.linktext+]</a>[+wf.wrapper+]</li>`
	!]
```

**Comment by modxuser** [comment]
I think that might be it - WOW, just a mo

**Comment by fourroses666** [comment]
Maybe it needs an extra outer;
&outerTpl=`@CODE:<ol class="ug-toc"><li><ol class="ug-toc">[+wf.wrapper+]</ol></li></ol>`

Not sure this is needed for active one; &hereClass=`current`

**Comment by modxuser** [comment]
Looking good - not 100%, but a lot better than what I had.

I now realise what mistake(s) I made - as I created the pages I copied the rendered source code - which has all of the classes defined

But, some of the classes are created via the Javascript - now, not having to add them manually, it makes a lot more sense and is easier. Thanks Marc

**Comment by modxuser** [comment]
The only thing that doesnt work is: when you click on "Getting Started" from [this page](https://docs.modx.com/evolution/1.0), the "Getting Started" menu isn't expanded

Local
![local](https://cloud.githubusercontent.com/assets/1859868/22151198/51b86484-df1d-11e6-9b4c-254f55f0482f.jpg)

Live
![live](https://cloud.githubusercontent.com/assets/1859868/22151204/58df48b8-df1d-11e6-95e8-9598953de49d.jpg)

**Comment by fourroses666** [comment]
We need a central location to place docs. There are too many.
When someone adds a new plugin or module etc. it can be best placed on one location.
Github would be the best place I guess so there can be created Pull requests etc.
That are my thoughts.

**Comment by fourroses666** [comment]
hmm need to check the live site

**Comment by modxuser** [comment]
Thats true - but they have to be created first

**Comment by fourroses666** [comment]
Does this work? 
```
&parentRowTpl=`@CODE:<li class="open"><i class="icon-caret-right"></i><a href="[+wf.link+]">[+wf.linktext+]</a>[+wf.wrapper+]</li>`
```

**Comment by modxuser** [comment]
No - that opens all of the menu items

**Comment by modxuser** [comment]
Looking at the rendered code:

Default:

    <ol class="ug-toc"><li><ol class="ug-toc"><li class="current"><i class="icon-caret-right"></i><a href="/index.php?id=103">Getting Started</a>

After the arrow icon is clicked:

    <ol class="ug-toc"><li><ol class="ug-toc"><li class="current open"><i class="icon-caret-down"></i><a href="/index.php?id=103">Getting Started</a>

The difference being

    <li class="current open">

**Comment by fourroses666** [comment]
So this didn't work?
```
[!Wayfinder? 
&startId=`0` 
&level=`5` 
&outerTpl=`@CODE:<ol class="ug-toc">[+wf.wrapper+]</ol>`
&rowTpl=`@CODE:<li[+wf.classes+]><a href="[+wf.link+]" [+wf.attributes+]>[+wf.linktext+]</a>[+wf.wrapper+]</li>`
&innerTpl=`@CODE:<ol class="ug-toc">[+wf.wrapper+]</ol>`
&innerRowTpl=`@CODE:<li[+wf.classes+]><a href="[+wf.link+]">[+wf.linktext+]</a></li>`
&parentRowTpl=`@CODE:<li><i class="icon-caret-right"></i><a href="[+wf.link+]">[+wf.linktext+]</a>[+wf.wrapper+]</li>`
&activeParentRowTpl=`@CODE:<li class="open"><i class="icon-caret-right"></i><a href="[+wf.link+]">[+wf.linktext+]</a>[+wf.wrapper+]</li>`
!]
```

**Comment by modxuser** [comment]
Marc, don't stress yourself

The icon is clickable and the function expands the menu item - the child pages are also in the main content area of the parent.

So all pages are reachable, in one way or another

**Comment by fourroses666** [comment]
If it didn't we might need somekind of parant parant active row, which might need JS to add the open class.

**Comment by modxuser** [comment]
No, that code doesnt work, this works to 98%

    [[Wayfinder? 
    &startId=`2` 
    &outerTpl=`@CODE:<ol class="ug-toc"><li><ol class="ug-toc">[+wf.wrapper+]</ol></li></ol>` 
    &rowTpl=`@CODE:<li[+wf.classes+]><a href="[+wf.link+]" [+wf.attributes+]>[+wf.linktext+]</a>[+wf.wrapper+]</li>` 
    &innerTpl=`@CODE:<ol class="ug-toc">[+wf.wrapper+]</ol>` 
    &innerRowTpl=`@CODE:<li[+wf.classes+]><a href="[+wf.link+]">[+wf.linktext+]</a></li>` 
    &parentRowTpl=`@CODE:<li><i class="icon-caret-right"></i><a href="[+wf.link+]">[+wf.linktext+]</a>[+wf.wrapper+]</li>` 
    &activeParentRowTpl=`@CODE:<li class="current"><i class="icon-caret-right"></i><a href="[+wf.link+]">[+wf.linktext+]</a>[+wf.wrapper+]</li>` 
    &hideSubMenus=`0` 
    &removeNewLines=`1`
    ]]

**Comment by modxuser** [comment]
Thanks for your help Marc :)

I will now crack on with getting the rest sorted. Have a nice one.

**Comment by fourroses666** [comment]
Didn't see your reply @dmi3yy 
https://github.com/modxcms/evolution/issues/1106#issuecomment-274066826

Yea thats perfect.

**Comment by modxuser** [comment]
@fourroses666 

Success, Marc it now works to 100%

Here the updated code:

    [[Wayfinder? 
    &startId=`2` 
    &outerTpl=`@CODE:<ol class="ug-toc"><li><ol class="ug-toc">[+wf.wrapper+]</ol></li></ol>` 
    &rowTpl=`@CODE:<li[+wf.classes+]><a href="[+wf.link+]" [+wf.attributes+]>[+wf.linktext+]</a>[+wf.wrapper+]</li>` 
    &innerTpl=`@CODE:<ol class="ug-toc">[+wf.wrapper+]</ol>` 
    &innerRowTpl=`@CODE:<li[+wf.classes+]><a href="[+wf.link+]">[+wf.linktext+]</a></li>` 
    &parentRowTpl=`@CODE:<li><i class="icon-caret-right" title="Click To Open or Close"></i><a href="[+wf.link+]">[+wf.linktext+]</a>[+wf.wrapper+]</li>` 
    &activeParentRowTpl=`@CODE:<li [+wf.classes+]><i class="icon-caret-down" title="Click To Open or Close"></i><a href="[+wf.link+]">[+wf.linktext+]</a>[+wf.wrapper+]</li>` 
    &parentClass=`open` 
    &hereClass=`current` 
    &lastClass=`` 
    &hideSubMenus=`0`
    ]]

**Comment by fourroses666** [comment]
Ah, nice one! 
```
&parentClass=`open` 
&hereClass=`current`
```

**Comment by modxuser** [comment]
Yeh, with slight modifications to the "active parent row" template as well:

From
    &activeParentRowTpl=`@CODE:<li class="current"><i class="icon-caret-right">

To
    &activeParentRowTpl=`@CODE:<li [+wf.classes+]><i class="icon-caret-down">

**Comment by zabudkin** [comment]
>modxuser commented 2 days ago
>If anyone wants the current system - I can zip it and post here, you can then upload
>to your local server and test - just a n idea
Good idea!
zip it and post here!

**Comment by modxuser** [comment]
@zabudkin 

The docs are finsihed and as such, I see no need to zip them anymore - they will be online shortly

**Comment by zabudkin** [comment]
@modxuser 
how long? )

**Comment by modxuser** [comment]
@zabudkin 

As already mentioned, the docs will be online shortly

I am currently adding more content and updating all of the images that are in the docs - this takes time.

You can still use the docs found on the [MODX website](https://docs.modx.com/evolution/1.0)

**Comment by modxuser** [comment]
@Deesen @yama @dmi3yy @pmfx @fourroses666 @bossloper @Nicola1971 @zabudkin @Tourist4Life @nick0

Due to unforseen circumstances I have to put the work on the Docs on hold, **BUT:**

The [English Docs site is online](http://www.evolution-docs.com/) (the server is not the fastest, but cheap LOL), but it is obviously a "work in progress"

I have duplicated the official docs, cleaned them up and modified them.

The HTML + CSS have had some major changes made - it seems that in each design move, the old code was always left in. I have (as best I could in the time) removed as much as possible. There is still redundant code, but that will be removed in time.

I have completely re-built the section: [Content Editing / Editing Resources](http://www.evolution-docs.com/documentation/content-editing/editing-resources/), it is now cleaner, better to understnad and all the images use the latest design + structure

Various other pages have been updated and grammatically corrected

I have also added a few new pages from the Wiki and from the MODX Wiki


@yama

Please take a closer look at [Modifiers](http://www.evolution-docs.com/documentation/designing/modifiers/) and [Cross References](http://www.evolution-docs.com/documentation/designing/cross-references)

I have added an extra page for the [@ modifiers](http://www.evolution-docs.com/documentation/designing/modifiers/at-modifiers) - more for clarity as anything else

Please let me know if thats OK with you

**Comment by pmfx** [comment]
@modxuser amazing work, you're the man! Thank you. Full Documentation updated with features added lately is so much important and should be a priority.

**Comment by pmfx** [comment]
http://www.evolution-docs.com/documentation/designing/document-caching
Part about dynamic content can be changed a little. Dynamic content based on GET parameters can be safely cached since we have this option in Configuration:

![2017-01-24-15-22-05](https://cloud.githubusercontent.com/assets/10888055/22250871/ed9d8464-e248-11e6-9fec-805bde0c8da6.png)

**Comment by modxuser** [comment]
There is still a lot to be changed - loads of images, due to the interface change and there is a lot of incorrect content as well.

All in time :)

**Comment by zabudkin** [comment]
@modxuser 
Please add this example (i'm get example from http://modx.im/blog/questions/4961.html)
to docs (section: Modifiers) (sorry for my bad english):

---------------------------------
Example with snippets:
<@IF: [!ip:substr(0,3):is('10.'):or:substr(0,8):is('192.168.')!]>
Ip address begin from 10. or ip address begin from 192.168.
<@ENDIF>
Where:
    - ip is snippet that return ip address of visiter.
---------------------------------

**Comment by yama** [comment]
[!$_SERVER['REMOTE_ADDR']:substr(0,3):is('10.'):or:substr(0,8):is('192.168.')!]

**Comment by Deesen** [comment]
http://web.archive.org/web/20160506161117/http://wiki.modxcms.com/index.php/PHx

**Comment by kivamp** [comment]
[!ip!]
```
<?php
return getenv('HTTP_CLIENT_IP')?:
getenv('HTTP_X_FORWARDED_FOR')?:
getenv('HTTP_X_FORWARDED')?:
getenv('HTTP_FORWARDED_FOR')?:
getenv('HTTP_FORWARDED')?:
getenv('REMOTE_ADDR');
```

**Comment by em-piguet** [comment]
I would like to help making a clear documentation but what about using a real tool for the job ? 
Maybe have a look at https://readthedocs.org/ ? Integration with github make it really easy to contribute and help... or something similar ?

**Comment by modxuser** [comment]
Just an update:

To those who have offered to help with the docs, thanks for the offer of help, but at the moment, it's like how the saying goes "Too many cooks spoil the broth".

Since I started on the Docs, I have seen that the codebase has rapidly declined in quality due to so many different people adding to the docs + extras docs.

**An example:**
Last night I spent 3 hours cleaning up and modifying the two Wayfinder pages that are in the extras section. I ended up having to go through Archive.org and cross reference the information that is relevant to Evo as the current docs text is a mix of Evo + Revo. The pages have multiple duplicate anchors, dead links and bad code

I am trying my best to get a codebase down that can be used in the future as a "Coding Standard".

Once all of the docs are duplicated, modified and all code is clean - then and only then will I let others modify / add to the site

**Comment by modxuser** [comment]
@Deesen 

PHx documentation can now be found here: [PHx](http://www.evolution-docs.com/extras/phx/)

**Comment by nick0** [comment]
Super work with the docs @modxuser.

I just had a quick look at the "PHx documentation" - fantastic work.

One thing might need slight rewording on the PHx Custom Modifiers page as Ditto does have an Outer Wrapper now...

```
phx:outer
    description: Surround not empty string with text.
    usage: [string:outer=`before|after`]
    example: Ditto has no outer template, but you can use this [+phx:input=`[!Ditto? &noResults=``!]`:outer=`<div class="xyz">|</div>`+]
    added by: Jako
```

Irrespective of Ditto's Outer Wrapper Template, the PHx outer modifier does come in handy sometimes when using MultiTV in Ditto templates

**Comment by nick0** [comment]
I would love to see a comparison page in the docs with equivalent calls of PHx, IF and the new built-in modifiers. At this stage I don't really know how to use the new built-in modifiers or its parameters etc, so I am unsure how to go about replacing PHx with the new built-in modifiers.

**Comment by pmfx** [comment]
@modxuser great work!
@nick0 100% agree with filters comparison page.

**Comment by modxuser** [comment]
@nick0 @pmfx  Thanks

I agree with the modifiers problem.

If people add their PHx calls, we can all have a go at converting them to IF + Built-In modifiers.

If can be [found here](https://forums.modx.com/thread/?thread=80876&page=1), it is extremely similar to PHx - I found converting my PHx calss very easy, one or two I wasn't sure about and asked @dmi3yy ,as can be seen in [the forum](https://forums.modx.com/thread/80876/snippet-if-for-modx-evo#dis-post-481754)

**Comment by nick0** [comment]
> If people add their PHx calls, we can all have a go at converting them to "IF" + "Built-In modifiers".

Great idea. Where would you like these posted? In Github here as a new issues page to start with?

**Comment by modxuser** [comment]
@nick0 

I think the discussion about PHx, If and the new Moidifiers is a sore at the moment and needs to be resolved ASAP

If we do the conversions, I think the best place would be in the Forum - more room and hopefully, more eyes.

**Comment by nick0** [comment]
Thanks @modxuser. I thought the Evo forums at Modx were dead but ok no worries. I will start compiling some examples first

**Comment by modxuser** [comment]
**UPDATE:**

Documentation site has been updated - It now contains "ALL" of the English Evo content from the MODX LLC documentation site and a hell of a lot more

**Latest:**
[More Extras](http://www.evolution-docs.com/extras/)
[More Tutorials](http://www.evolution-docs.com/tutorials/extras-how-to/)
Updated Documentation (both Evo + Extras) - multiple pages
Cleaner Code: which includes cleaning up "pre" tags by using the correct display brush and changing symbols to their equivalent numbers - as quite a few weren't showing in the front-end due to Evo rendering them

**Comment by risingisland** [comment]
Awesome work! Very useful/helpful.

**Comment by Nicola1971** [comment]
@modxuser great work. thank you very much

**Comment by zabudkin** [comment]
"PHx is out of the box". Where is modifiers? NOTHING about it.
"This is another doc-project about nothing new."

**Comment by Nicola1971** [comment]
@zabudkin http://www.evolution-docs.com/documentation/designing/modifiers/ ?

**Comment by modxuser** [comment]
@zabudkin 

Whats your problem ?

If you have documentation for PHx, then please feel free to send it to me / post it here or post it on the forums

> This is another doc-project about nothing new

If you have a better idea, then feel free to tell me / the community - I have invested hundreds of hours of work for the current site, so refering to it as "nothing new" is absolutely wrong - there is a lot more information in the site and it all works / is now displayed properly

If you think you can do something better - feel free :)

**Comment by pmfx** [comment]
Great work. Just a note there is a PHx Custom Modifier "phx:outer" saying that Ditto has no outer template. It does.

![screenshot_1](https://cloud.githubusercontent.com/assets/10888055/23143096/c4a0e4de-f7bf-11e6-9ce3-572d13c148db.jpg)

It is called "outerTpl" and would be nice to add it to Ditto Template parameters documentation.

**Comment by modxuser** [comment]
As @nick0 describes here: https://github.com/modxcms/evolution/issues/1106#issuecomment-279576334

I have updated [PHx Custom Modifiers](http://www.evolution-docs.com/extras/phx/phx-custom-modifiers) + [Ditto Template Parameters](http://www.evolution-docs.com/extras/ditto/ditto-parameters), although I am not sure about the text for the "[outerTpl](http://www.evolution-docs.com/extras/ditto/ditto-parameters#template-parameters)"

I have also added anchors in the [PHx](http://www.evolution-docs.com/extras/phx/) page for quicker access to the different sections, including [modifiers](http://www.evolution-docs.com/extras/phx/#phx-modifiers)

**Comment by nick0** [comment]
Great work @modxuser.

I wonder if it would be worthwhile having example usage for the outerTpl so people know how to use it?
Here is an example:
`
<div class="grid">
 <ul>
  [+wrapper+]
 </ul>
</div>
`

Maybe in the ditto section we could have an Examples page as the 4th ditto page? This way we could show Ditto Calls with Outer Template , Row Template, Filtering etc

If you would like this, let me know and I will shoot through some examples to get things started. Let me know where you want it - eg here, via email etc.

**Comment by fourroses666** [comment]
or `@code:<div class="grid"><ul>((wrapper))</ul></div>`

**Comment by modxuser** [comment]
@nick0 

Feel free to create your code using HTML and send me that code

I have setup a temp email for this stuff: noreply@evolution-docs.com

The same goes with PHx calls, lets see if we can't get something started or at least try LOL

**Comment by nick0** [comment]
Whoops, just sent you the text documents via email and came here to reply to let you know and I see you have edited your post. Wow, you deserve a medal for dealing with that to build the docs :)
Ok, I did the find and replace and sent through again. Fingers crossed I got it right.

**Comment by modxuser** [comment]
@nick0 

I have just sent you an email with an updated version of your Ditto calls, please take a look and compare your file against the one attached in my mail

**Comment by nick0** [comment]
Thanks @modxuser. The docs look great.
RSS feed updated, cheers.
ps I have started that post in the forum for the [PHx / IF / Modifier Call Comparisons]([url]
https://forums.modx.com/thread/101774/phx-if-modifier-call-comparisons#dis-post-548897(url))

**Comment by modxuser** [comment]
@nick0 sorry for the late reply, am currently up to my neck in my normal work and will be for the next few weeks

**Comment by nick0** [comment]
No worries @modxuser, thanks for the update

**Comment by modxuser** [comment]
Just some feedback on this:

My work and my "BA" now take much more of my time and as such I have given up the documentation site as I simply can no-longer afford the time.

The website including domain has been officially transfered over to Dmi3yy, he was the only person to answer the request of ownership transfer.

Hosting and maintenance is now in his hands - what he does with the site and whether or not he gives other community members manager permissions is something I can not answer.

Hopefully the English documentation will carry on.
